src/irmin-containers/lww_register.ml

```diff
@@ -0,0 +1,105 @@
+(*
+ * Copyright (c) 2020 KC Sivaramakrishnan <kc@kcsrk.info>
+ * Copyright (c) 2020 Anirudh Sunder Raj <anirudh6626@gmail.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open Lwt.Infix
+
+let return = Lwt.return
+
+let info = Irmin_unix.info
+
+module type Input = sig
+  include Irmin.Type.S
+
+  val compare : t -> t -> int
+end
+
+module Time : sig
+  type t
+
+  val t : t Irmin.Type.t
+
+  val compare : t -> t -> int
+
+  val get_time : unit -> t
+end = struct
+  type t = int64
+
+  let t = Irmin.Type.int64
+
+  let compare = Int64.compare
+
+  let get_time () = Unix.gettimeofday () |> Int64.of_float
+end
+
+module LWW (V : Input) : Irmin.Contents.S with type t = V.t * Time.t = struct
+  type t = V.t * Time.t
+
+  let t = Irmin.Type.(pair V.t Time.t)
+
+  let compare (v1, t1) (v2, t2) =
+    let res = Time.compare t1 t2 in
+    if res = 0 then V.compare v1 v2 else res
```

Comment:
It seemed like a good idea then to expect an explicit comparison for the values. On second thoughts, the choice of picking the bigger value is as arbitrary as picking `v1` all the time. You'll just need to explain in the documentation what the semantics is and mention that if the timestamps are the same, then one of the values is arbitrarily chosen. Then `Input` is not necessary. It could just be `Irmin.Type.S`.

Reply:
The comparison function for value types is necessary for the merge to converge.